2025-03-20 19:30:15,521 - __main__ - INFO - Loading data...
2025-03-20 19:30:18,260 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 19:30:18,260 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:649: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:649: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 19:30:20,728 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 19:30:21,493 - __main__ - INFO - Training on: cuda
2025-03-20 19:30:21,494 - __main__ - INFO - Starting training...
2025-03-20 19:30:50,892 - __main__ - INFO - Saved best model with validation loss: 0.3209
2025-03-20 19:31:17,539 - __main__ - INFO - Saved best model with validation loss: 0.2991
2025-03-20 19:31:43,255 - __main__ - INFO - Saved best model with validation loss: 0.2799
2025-03-20 19:32:10,340 - __main__ - INFO - Saved best model with validation loss: 0.2245
2025-03-20 19:32:38,194 - __main__ - INFO - Saved best model with validation loss: 0.1879
2025-03-20 19:33:04,731 - __main__ - INFO - Saved best model with validation loss: 0.1769
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 857, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 791, in main
    metrics = trainer.train(
  File "/app/nse/nn/perseptron_test.py", line 529, in train
    tb_writer.add_histogram('grad/%s'%tag, param.grad.data.cpu().numpy(), global_step = epoch*(int(len(dataloader)/batch_size))+idx + 1)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py", line 499, in add_histogram
    histogram(tag, values, bins, max_bins=max_bins), global_step, walltime
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/summary.py", line 485, in histogram
    hist = make_histogram(values.astype(float), bins, max_bins)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/summary.py", line 530, in make_histogram
    raise ValueError("The histogram is empty, please file a bug report.")
ValueError: The histogram is empty, please file a bug report.
2025-03-20 21:05:43,839 - __main__ - INFO - Loading data...
2025-03-20 21:06:03,246 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 21:06:03,246 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:649: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:649: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 21:06:06,366 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 21:06:07,985 - __main__ - INFO - Training on: cuda
2025-03-20 21:06:07,986 - __main__ - INFO - Starting training...
2025-03-20 21:06:50,415 - __main__ - INFO - Saved best model with validation loss: 0.0787
2025-03-20 21:07:34,728 - __main__ - INFO - Saved best model with validation loss: 0.0713
2025-03-20 21:08:13,180 - __main__ - INFO - Saved best model with validation loss: 0.0704
2025-03-20 21:12:23,876 - __main__ - INFO - Epoch [10/100], Train Loss: 0.0270, Val Loss: 0.0929, LR: 0.000050
2025-03-20 21:18:27,080 - __main__ - INFO - Epoch [20/100], Train Loss: 0.0197, Val Loss: 0.1250, LR: 0.000025
2025-03-20 21:20:13,297 - __main__ - INFO - Early stopping triggered at epoch 23
2025-03-20 21:20:14,251 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 21:20:14,251 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 21:20:15,047 - __main__ - INFO - Loaded best model with validation loss: 0.07039611786603928
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 857, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 815, in main
    plot_eda(pd.DataFrame(y_pred), config['output_dir'] + "/target_pred_transform.png")
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py", line 778, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 114, in arrays_to_mgr
    index = _extract_index(arrays)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 667, in _extract_index
    raise ValueError("If using all scalar values, you must pass an index")
ValueError: If using all scalar values, you must pass an index
2025-03-20 21:34:15,632 - __main__ - INFO - Loading data...
2025-03-20 21:34:39,073 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 21:34:39,073 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 21:34:42,324 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 21:34:43,827 - __main__ - INFO - Training on: cuda
2025-03-20 21:34:43,827 - __main__ - INFO - Starting training...
2025-03-20 21:35:03,822 - __main__ - INFO - Saved best model with validation loss: 0.0971
2025-03-20 21:35:22,931 - __main__ - INFO - Saved best model with validation loss: 0.0784
2025-03-20 21:35:42,930 - __main__ - INFO - Saved best model with validation loss: 0.0753
2025-03-20 21:36:02,281 - __main__ - INFO - Saved best model with validation loss: 0.0742
2025-03-20 21:36:22,526 - __main__ - INFO - Saved best model with validation loss: 0.0726
2025-03-20 21:36:41,836 - __main__ - INFO - Saved best model with validation loss: 0.0713
2025-03-20 21:37:54,397 - __main__ - INFO - Epoch [10/100], Train Loss: 0.0000, Val Loss: 0.0767, LR: 0.000100
2025-03-20 21:40:57,186 - __main__ - INFO - Epoch [20/100], Train Loss: 0.0000, Val Loss: 0.0870, LR: 0.000025
2025-03-20 21:42:47,048 - __main__ - INFO - Early stopping triggered at epoch 26
2025-03-20 21:42:47,914 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 21:42:47,914 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 21:42:49,119 - __main__ - INFO - Loaded best model with validation loss: 0.07129181921482086
2025-03-20 21:42:49,305 - __main__ - INFO - Initializing model for c_delta_y...
2025-03-20 21:42:49,309 - __main__ - INFO - Training on: cuda
2025-03-20 21:42:49,309 - __main__ - INFO - Starting training...
2025-03-20 21:43:08,136 - __main__ - INFO - Saved best model with validation loss: 4.0238
2025-03-20 21:45:53,896 - __main__ - INFO - Epoch [10/100], Train Loss: 0.0003, Val Loss: 5.5000, LR: 0.000050
2025-03-20 21:48:59,319 - __main__ - INFO - Epoch [20/100], Train Loss: 0.0002, Val Loss: 6.2764, LR: 0.000013
2025-03-20 21:49:17,586 - __main__ - INFO - Early stopping triggered at epoch 21
2025-03-20 21:49:19,531 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 21:49:19,531 - __main__ - INFO - Evaluating model for c_delta_y...
2025-03-20 21:49:19,919 - __main__ - INFO - Loaded best model with validation loss: 4.023841381072998
2025-03-20 21:49:20,038 - __main__ - INFO - Initializing model for c_std_z...
2025-03-20 21:49:20,050 - __main__ - INFO - Training on: cuda
2025-03-20 21:49:20,051 - __main__ - INFO - Starting training...
2025-03-20 21:49:39,059 - __main__ - INFO - Saved best model with validation loss: 0.1825
2025-03-20 21:49:57,806 - __main__ - INFO - Saved best model with validation loss: 0.1167
2025-03-20 21:50:16,739 - __main__ - INFO - Saved best model with validation loss: 0.0751
2025-03-20 21:50:35,585 - __main__ - INFO - Saved best model with validation loss: 0.0512
2025-03-20 21:50:54,521 - __main__ - INFO - Saved best model with validation loss: 0.0427
2025-03-20 21:51:13,084 - __main__ - INFO - Saved best model with validation loss: 0.0374
2025-03-20 21:51:31,508 - __main__ - INFO - Saved best model with validation loss: 0.0343
2025-03-20 21:51:50,442 - __main__ - INFO - Saved best model with validation loss: 0.0329
2025-03-20 21:52:09,112 - __main__ - INFO - Saved best model with validation loss: 0.0314
2025-03-20 21:52:28,350 - __main__ - INFO - Saved best model with validation loss: 0.0312
2025-03-20 21:52:28,350 - __main__ - INFO - Epoch [10/100], Train Loss: 0.0000, Val Loss: 0.0312, LR: 0.000100
2025-03-20 21:52:47,499 - __main__ - INFO - Saved best model with validation loss: 0.0300
2025-03-20 21:53:06,142 - __main__ - INFO - Saved best model with validation loss: 0.0299
2025-03-20 21:53:24,740 - __main__ - INFO - Saved best model with validation loss: 0.0295
2025-03-20 21:53:43,613 - __main__ - INFO - Saved best model with validation loss: 0.0292
2025-03-20 21:54:02,229 - __main__ - INFO - Saved best model with validation loss: 0.0287
2025-03-20 21:54:20,835 - __main__ - INFO - Saved best model with validation loss: 0.0286
2025-03-20 21:54:39,353 - __main__ - INFO - Saved best model with validation loss: 0.0284
2025-03-20 21:55:16,314 - __main__ - INFO - Saved best model with validation loss: 0.0282
2025-03-20 21:55:34,912 - __main__ - INFO - Epoch [20/100], Train Loss: 0.0000, Val Loss: 0.0286, LR: 0.000100
2025-03-20 21:58:39,556 - __main__ - INFO - Epoch [30/100], Train Loss: 0.0000, Val Loss: 0.0291, LR: 0.000050
2025-03-20 22:01:26,040 - __main__ - INFO - Early stopping triggered at epoch 39
2025-03-20 22:01:26,842 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:01:26,842 - __main__ - INFO - Evaluating model for c_std_z...
2025-03-20 22:01:27,662 - __main__ - INFO - Loaded best model with validation loss: 0.02815156616270542
2025-03-20 22:01:27,801 - __main__ - INFO - Initializing model for c_std_y...
2025-03-20 22:01:27,806 - __main__ - INFO - Training on: cuda
2025-03-20 22:01:27,806 - __main__ - INFO - Starting training...
2025-03-20 22:01:46,952 - __main__ - INFO - Saved best model with validation loss: 0.3618
2025-03-20 22:02:05,927 - __main__ - INFO - Saved best model with validation loss: 0.2497
2025-03-20 22:02:24,948 - __main__ - INFO - Saved best model with validation loss: 0.2239
2025-03-20 22:02:43,745 - __main__ - INFO - Saved best model with validation loss: 0.2089
2025-03-20 22:03:02,681 - __main__ - INFO - Saved best model with validation loss: 0.1983
2025-03-20 22:03:21,510 - __main__ - INFO - Saved best model with validation loss: 0.1951
2025-03-20 22:03:40,435 - __main__ - INFO - Saved best model with validation loss: 0.1920
2025-03-20 22:04:36,011 - __main__ - INFO - Epoch [10/100], Train Loss: 0.0000, Val Loss: 0.1959, LR: 0.000100
2025-03-20 22:07:42,288 - __main__ - INFO - Epoch [20/100], Train Loss: 0.0000, Val Loss: 0.2062, LR: 0.000025
2025-03-20 22:09:51,880 - __main__ - INFO - Early stopping triggered at epoch 27
2025-03-20 22:09:52,623 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:09:52,623 - __main__ - INFO - Evaluating model for c_std_y...
2025-03-20 22:09:53,294 - __main__ - INFO - Loaded best model with validation loss: 0.1919747292995453
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 855, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 813, in main
    plot_eda(pd.DataFrame(y_pred), config['output_dir'] + "/target_pred_transform.png")
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py", line 778, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 114, in arrays_to_mgr
    index = _extract_index(arrays)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 667, in _extract_index
    raise ValueError("If using all scalar values, you must pass an index")
ValueError: If using all scalar values, you must pass an index
2025-03-20 22:23:17,453 - __main__ - INFO - Loading data...
2025-03-20 22:23:32,273 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 22:23:32,274 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 22:23:34,776 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 22:23:35,901 - __main__ - INFO - Training on: cuda
2025-03-20 22:23:35,901 - __main__ - INFO - Starting training...
2025-03-20 22:23:54,844 - __main__ - INFO - Saved best model with validation loss: 0.0971
2025-03-20 22:24:13,354 - __main__ - INFO - Saved best model with validation loss: 0.0784
2025-03-20 22:24:14,211 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:24:14,211 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 22:24:14,304 - __main__ - INFO - Loaded best model with validation loss: 0.07837457209825516
2025-03-20 22:24:14,541 - __main__ - INFO - Initializing model for c_delta_y...
2025-03-20 22:24:14,555 - __main__ - INFO - Training on: cuda
2025-03-20 22:24:14,555 - __main__ - INFO - Starting training...
2025-03-20 22:24:33,318 - __main__ - INFO - Saved best model with validation loss: 4.0168
2025-03-20 22:24:52,588 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:24:52,588 - __main__ - INFO - Evaluating model for c_delta_y...
2025-03-20 22:24:53,057 - __main__ - INFO - Loaded best model with validation loss: 4.016848564147949
2025-03-20 22:24:53,178 - __main__ - INFO - Initializing model for c_std_z...
2025-03-20 22:24:53,182 - __main__ - INFO - Training on: cuda
2025-03-20 22:24:53,182 - __main__ - INFO - Starting training...
2025-03-20 22:25:12,100 - __main__ - INFO - Saved best model with validation loss: 0.1726
2025-03-20 22:25:30,925 - __main__ - INFO - Saved best model with validation loss: 0.1169
2025-03-20 22:25:31,725 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:25:31,725 - __main__ - INFO - Evaluating model for c_std_z...
2025-03-20 22:25:31,812 - __main__ - INFO - Loaded best model with validation loss: 0.11685731261968613
2025-03-20 22:25:31,929 - __main__ - INFO - Initializing model for c_std_y...
2025-03-20 22:25:31,933 - __main__ - INFO - Training on: cuda
2025-03-20 22:25:31,933 - __main__ - INFO - Starting training...
2025-03-20 22:25:50,749 - __main__ - INFO - Saved best model with validation loss: 0.3360
2025-03-20 22:26:09,644 - __main__ - INFO - Saved best model with validation loss: 0.2460
2025-03-20 22:26:10,373 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:26:10,373 - __main__ - INFO - Evaluating model for c_std_y...
2025-03-20 22:26:10,459 - __main__ - INFO - Loaded best model with validation loss: 0.2459951788187027
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 855, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 813, in main
    plot_eda(pd.DataFrame(y_pred), config['output_dir'] + "/target_pred_transform.png")
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py", line 778, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 114, in arrays_to_mgr
    index = _extract_index(arrays)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 664, in _extract_index
    raise ValueError("Per-column arrays must each be 1-dimensional")
ValueError: Per-column arrays must each be 1-dimensional
2025-03-20 22:27:26,623 - __main__ - INFO - Loading data...
2025-03-20 22:27:36,084 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 22:27:36,084 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 22:27:38,736 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 22:27:39,516 - __main__ - INFO - Training on: cuda
2025-03-20 22:27:39,516 - __main__ - INFO - Starting training...
2025-03-20 22:27:59,124 - __main__ - INFO - Saved best model with validation loss: 0.0971
2025-03-20 22:28:17,501 - __main__ - INFO - Saved best model with validation loss: 0.0784
2025-03-20 22:28:18,305 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:28:18,305 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 22:28:18,400 - __main__ - INFO - Loaded best model with validation loss: 0.07837457209825516
2025-03-20 22:28:18,523 - __main__ - INFO - Initializing model for c_delta_y...
2025-03-20 22:28:18,529 - __main__ - INFO - Training on: cuda
2025-03-20 22:28:18,529 - __main__ - INFO - Starting training...
2025-03-20 22:28:37,417 - __main__ - INFO - Saved best model with validation loss: 4.0168
2025-03-20 22:28:57,162 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:28:57,162 - __main__ - INFO - Evaluating model for c_delta_y...
2025-03-20 22:28:57,503 - __main__ - INFO - Loaded best model with validation loss: 4.016848564147949
2025-03-20 22:28:57,623 - __main__ - INFO - Initializing model for c_std_z...
2025-03-20 22:28:57,630 - __main__ - INFO - Training on: cuda
2025-03-20 22:28:57,630 - __main__ - INFO - Starting training...
2025-03-20 22:29:16,861 - __main__ - INFO - Saved best model with validation loss: 0.1726
2025-03-20 22:29:35,685 - __main__ - INFO - Saved best model with validation loss: 0.1169
2025-03-20 22:29:36,695 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:29:36,695 - __main__ - INFO - Evaluating model for c_std_z...
2025-03-20 22:29:36,805 - __main__ - INFO - Loaded best model with validation loss: 0.11685731261968613
2025-03-20 22:29:36,923 - __main__ - INFO - Initializing model for c_std_y...
2025-03-20 22:29:36,927 - __main__ - INFO - Training on: cuda
2025-03-20 22:29:36,927 - __main__ - INFO - Starting training...
2025-03-20 22:29:55,962 - __main__ - INFO - Saved best model with validation loss: 0.3360
2025-03-20 22:30:14,920 - __main__ - INFO - Saved best model with validation loss: 0.2460
2025-03-20 22:30:15,645 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:30:15,645 - __main__ - INFO - Evaluating model for c_std_y...
2025-03-20 22:30:15,730 - __main__ - INFO - Loaded best model with validation loss: 0.2459951788187027
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
(459344, 1)
(459344, 1)
(459344, 1)
(459344, 1)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 855, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 813, in main
    plot_eda(pd.DataFrame(y_pred), config['output_dir'] + "/target_pred_transform.png")
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py", line 778, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 114, in arrays_to_mgr
    index = _extract_index(arrays)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/internals/construction.py", line 664, in _extract_index
    raise ValueError("Per-column arrays must each be 1-dimensional")
ValueError: Per-column arrays must each be 1-dimensional
2025-03-20 22:31:28,956 - __main__ - INFO - Loading data...
2025-03-20 22:31:32,299 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 22:31:32,299 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 22:31:34,765 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 22:31:35,862 - __main__ - INFO - Training on: cuda
2025-03-20 22:31:35,862 - __main__ - INFO - Starting training...
2025-03-20 22:31:55,303 - __main__ - INFO - Saved best model with validation loss: 0.0971
2025-03-20 22:31:56,039 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:31:56,039 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 22:31:56,162 - __main__ - INFO - Loaded best model with validation loss: 0.09714733064174652
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
Traceback (most recent call last):
  File "/app/nse/nn/perseptron_test.py", line 855, in <module>
    main()
  File "/app/nse/nn/perseptron_test.py", line 810, in main
    y_pred[target] = np.squeezy(np.array(y_pred_tmp))
  File "/usr/local/lib/python3.10/dist-packages/numpy/__init__.py", line 320, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'squeezy'. Did you mean: 'squeeze'?
2025-03-20 22:32:25,444 - __main__ - INFO - Loading data...
2025-03-20 22:32:28,322 - __main__ - INFO - Loaded data from /app/nse/ml/output_19_01_2025_2: X shape=(2322432, 10), y shape=(2322432, 4)
2025-03-20 22:32:28,323 - __main__ - INFO - Preprocessing and splitting data...
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 22:32:30,898 - __main__ - INFO - Initializing model for c_delta_z...
/usr/local/lib/python3.10/dist-packages/torch/optim/lr_scheduler.py:61: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
2025-03-20 22:32:31,849 - __main__ - INFO - Training on: cuda
2025-03-20 22:32:31,849 - __main__ - INFO - Starting training...
2025-03-20 22:32:51,420 - __main__ - INFO - Saved best model with validation loss: 0.0971
2025-03-20 22:32:52,198 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:32:52,198 - __main__ - INFO - Evaluating model for c_delta_z...
2025-03-20 22:32:52,286 - __main__ - INFO - Loaded best model with validation loss: 0.09714733064174652
2025-03-20 22:32:52,409 - __main__ - INFO - Initializing model for c_delta_y...
2025-03-20 22:32:52,414 - __main__ - INFO - Training on: cuda
2025-03-20 22:32:52,414 - __main__ - INFO - Starting training...
2025-03-20 22:33:11,787 - __main__ - INFO - Saved best model with validation loss: 4.0139
2025-03-20 22:33:12,504 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:33:12,504 - __main__ - INFO - Evaluating model for c_delta_y...
2025-03-20 22:33:12,603 - __main__ - INFO - Loaded best model with validation loss: 4.013875484466553
2025-03-20 22:33:12,733 - __main__ - INFO - Initializing model for c_std_z...
2025-03-20 22:33:12,738 - __main__ - INFO - Training on: cuda
2025-03-20 22:33:12,739 - __main__ - INFO - Starting training...
2025-03-20 22:33:31,752 - __main__ - INFO - Saved best model with validation loss: 0.1888
2025-03-20 22:33:32,420 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:33:32,420 - __main__ - INFO - Evaluating model for c_std_z...
2025-03-20 22:33:32,518 - __main__ - INFO - Loaded best model with validation loss: 0.1887904852628708
2025-03-20 22:33:32,636 - __main__ - INFO - Initializing model for c_std_y...
2025-03-20 22:33:32,640 - __main__ - INFO - Training on: cuda
2025-03-20 22:33:32,640 - __main__ - INFO - Starting training...
2025-03-20 22:33:52,213 - __main__ - INFO - Saved best model with validation loss: 0.3436
2025-03-20 22:33:52,898 - __main__ - INFO - Training history plot saved to ./outputs_19_03_2025/training_history.png
2025-03-20 22:33:52,898 - __main__ - INFO - Evaluating model for c_std_y...
2025-03-20 22:33:52,994 - __main__ - INFO - Loaded best model with validation loss: 0.34357553720474243
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
/app/nse/nn/perseptron_test.py:647: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared.
  target.hist(bins=50, ax=ax)
2025-03-20 22:34:06,355 - __main__ - INFO - Final Test MSE Loss: 10757.9365
2025-03-20 22:34:19,001 - __main__ - INFO - Model saved to ./outputs_19_03_2025/regression_neural_network.pth
(2312849, 11) (2312849, 4)
Split shapes: train: X(1486776, 11), y(1486776, 4), test: X(459344, 11), y(459344, 4), valid: X(366729, 11), y(366729, 4)
(459344,)
(459344,)
(459344,)
(459344,)
                mse        r2      rmse
c_delta_z  0.116501  0.651813  0.341323
c_delta_y  4.410671  0.137427  2.100160
c_std_z    0.174047  0.822840  0.417189
c_std_y    0.340022  0.881212  0.583114
                   mse        r2       rmse
c_delta_z   955.484827  0.583771  30.910918
c_delta_y   822.415062  0.029359  28.677780
c_std_z    1280.676143  0.442466  35.786536
c_std_y    1104.218497  0.652871  33.229783
